runtime/src/accounts_db.rs

```diff
@@ -8552,11 +8555,10 @@ impl AccountsDb {
                     slots
                         .iter()
                         .filter_map(|slot| {
-                            if requested_slots.contains(slot)
-                                && (self.accounts_index.is_alive_root(*slot)
-                                    || ancestors
-                                        .map(|ancestors| ancestors.contains_key(slot))
-                                        .unwrap_or_default())
+                            if self.accounts_index.is_alive_root(*slot)
```

Comment:
nit & out of this PRs scope:

This is a confusing use of `filter_map`; we have an explicit `if {...} else { None }`. imo, this would be better suited as a `filter` (with this condition), then a `filter_map` w/ a modified body:
```rust
(!storages.is_empty()).then_some((storages, *slot))

Reply:
I have follow up prs which simplify this logic.